File: src/backend/Bennetr.BrickInv.Api/Bennetr.BrickInv.Api/Models/GroupInvite.cs
using System.ComponentModel.DataAnnotations;

namespace Bennetr.BrickInv.Api.Models;

public class GroupInvite
{
    [MaxLength(36)]
    public string Id { get; set; } = string.Empty;

    public DateTime Created { get; set; }

    public Group Group { get; set; } = null!;

    public UserProfile Issuer { get; set; } = null!;

    public UserProfile Recipient { get; set; } = null!;
}

File: src/frontend/src/domain/models/GroupInvite.ts
import { Instance, SnapshotIn, SnapshotOut, types } from 'mobx-state-tree';
import { MSTDateTime } from '../../utils';
import UserProfile from './UserProfile.ts';
import Group from './Group.ts';

const GroupInvite = types.model('GroupInvite', {
  id: types.identifier,
  created: MSTDateTime,
  updated: MSTDateTime,
  group: Group,
  issuer: UserProfile,
  recipient: UserProfile
});

export default GroupInvite;

export interface IGroupInvite extends Instance<typeof GroupInvite> {
}

export interface IGroupInviteSnapshotIn extends SnapshotIn<typeof GroupInvite> {
}

export interface IGroupInviteSnapshotOut extends SnapshotOut<typeof GroupInvite> {
}
